codec-http2/src/test/java/io/netty/handler/codec/http2/Http2HeadersTest.java

```diff
@@ -0,0 +1,83 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class Http2HeadersTest {
+
+    @Test
+    public void testGetPseudoHeader() {
+        for (PseudoHeaderName pseudoHeaderName : PseudoHeaderName.values()) {
+            assertSame(pseudoHeaderName, getPseudoHeader(pseudoHeaderName.value()));
+            assertSame(pseudoHeaderName, getPseudoHeader(new AsciiString(pseudoHeaderName.value().array())));
+            assertSame(pseudoHeaderName, getPseudoHeader(pseudoHeaderName.value().toString()));
+            assertSame(pseudoHeaderName, getPseudoHeader(new String(pseudoHeaderName.value().toCharArray())));
+            assertSame(pseudoHeaderName, getPseudoHeader(new StringBuilder(pseudoHeaderName.value())));
+        }
+    }
+
+    @Test
+    public void testIsPseudoHeader() {
+        // same as before but for isPseudoHeader
+        for (PseudoHeaderName pseudoHeaderName : PseudoHeaderName.values()) {
+            assertTrue(isPseudoHeader(pseudoHeaderName.value()));
+            assertTrue(isPseudoHeader(new AsciiString(pseudoHeaderName.value().array())));
+            assertTrue(isPseudoHeader(pseudoHeaderName.value().toString()));
+            assertTrue(isPseudoHeader(new String(pseudoHeaderName.value().toCharArray())));
+            assertTrue(isPseudoHeader(new StringBuilder(pseudoHeaderName.value())));
+        }
+    }
+
+    @Test
+    public void testPseudonameStringHashCodeIsValid() {
+        // do the same but throw a runtime exception here
+        // this is to ensure that the hashcode of the strings are the same as the hashcode of the enum
+        // convert the rest of the code is a proper JUnit test
+        assertEquals(METHOD.value().toString().hashCode(), METHOD_STRING_HASHCODE);
+        assertEquals(SCHEME.value().toString().hashCode(), SCHEME_STRING_HASHCODE);
+        assertEquals(STATUS.value().toString().hashCode(), STATUS_STRING_HASHCODE);
+        assertEquals(PATH.value().toString().hashCode(), PATH_STRING_HASHCODE);
+        assertEquals(AUTHORITY.value().toString().hashCode(), AUTHORITY_STRING_HASHCODE);
+        assertEquals(PROTOCOL.value().toString().hashCode(), PROTOCOL_STRING_HASHCODE);
+        // do the same with a new String, which hashcode is not computed yet
+        assertEquals(new String(METHOD.toString().toCharArray()), METHOD_STRING_HASHCODE);
+        assertEquals(new String(SCHEME.toString().toCharArray()), SCHEME_STRING_HASHCODE);
+        assertEquals(new String(STATUS.toString().toCharArray()), STATUS_STRING_HASHCODE);
+        assertEquals(new String(PATH.toString().toCharArray()), PATH_STRING_HASHCODE);
+        assertEquals(new String(AUTHORITY.toString().toCharArray()), AUTHORITY_STRING_HASHCODE);
+        assertEquals(new String(PROTOCOL.toString().toCharArray()), PROTOCOL_STRING_HASHCODE);
```

Comment:
@franz1981 I think you missed to call `hashCode()` on these ?

Reply:
Oh gosh, you are right :/